docs/modules/ROOT/pages/api/erc721.adoc

```diff
@@ -689,6 +689,102 @@ See <<IERC721-ApprovalForAll,IERC721::ApprovalForAll>>.
 
 See <<IERC721-Transfer,IERC721::Transfer>>.
 
+== Extensions
+
+[.contract]
+[[ERC721URIStorageComponent]]
+=== `++ERC721URIStorageComponent++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.14.0/src/token/erc721/extensions/erc721_uri_storage.cairo[{github-icon},role=heading-link]
+
+```cairo
+use openzeppelin::token::extensions::ERC721URIStorageComponent;
+```
+
+:MetadataUpdated: xref:ERC721URIStorageComponent-MetadataUpdated[MetadataUpdated]
+
+Extension of ERC721 to support dynamic NFTs.
+It is an implementation of <<IERC721Metadata, IERC721Metadata>> but with a different `token_uri` behavior.
+
+NOTE: Implementing xref:#ERC721Component[ERC721Component] is a requirement for this component to be implemented.
+
+This extension keeps a track of URIs of each token id of a particular ERC721 token. The URI of any `token_id` can be set by calling the 
+internal function, xref:#ERC721URIStorageComponent-set_token_uri[set_token_uri]. Updated `token_uri` can be queried through the external function  
+xref:#ERC721URIStorageComponent-token_uri[token_uri].
```

Comment:
What do you think of improving this section as mentioned in `#3` https://github.com/OpenZeppelin/cairo-contracts/pull/1031#discussion_r1667729192? If you agree with the suggested description above the component, we can maybe just reuse that here

Reply:
Okay I will reuse the description written above the component in code, after updating it with the suggested changes

Reply:
@andrew-fleming, does this seems good?
```The ERC721URIStorage component provides a flexible IERC721Metadata implementation that enables
storage-based token URI management. The URI of any `token_id` can be set by calling the 
internal function, xref:#ERC721URIStorageComponent-set_token_uri[set_token_uri]. Updated `token_uri` can be queried through the external function  
xref:#ERC721URIStorageComponent-token_uri[token_uri].```

Reply:
Can you please resolve some conversations, it becomes difficult to track what changes have been applied?